scripts/jenkins/groovy/defineTestStages.groovy

```diff
@@ -330,68 +315,99 @@ def call(final pipelineContext) {
       image: "${pipelineContext.getBuildConfig().DOCKER_REGISTRY}/opsh2oai/h2o-3/dev-r-3.5.3-jdk-11:${pipelineContext.getBuildConfig().DEFAULT_IMAGE_VERSION_TAG}"
     ],
     [
-      stageName: 'Java 10 JUnit', target: 'test-junit-10-jenkins', pythonVersion: '2.7', javaVersion: 10,
+      stageName: 'Java 14 JUnit', target: 'test-junit-14-jenkins', pythonVersion: '2.7', javaVersion: 14,
       timeoutValue: 180, component: pipelineContext.getBuildConfig().COMPONENT_JAVA, additionalTestPackages: [pipelineContext.getBuildConfig().COMPONENT_PY]
     ],
     [
-      stageName: 'Java 11 JUnit', target: 'test-junit-11-jenkins', pythonVersion: '2.7', javaVersion: 11,
-      timeoutValue: 180, component: pipelineContext.getBuildConfig().COMPONENT_JAVA, additionalTestPackages: [pipelineContext.getBuildConfig().COMPONENT_PY]
+      stageName: 'Py3.6 Single Node', target: 'test-pyunit-single-node', pythonVersion: '3.6',
```

Comment:
Py 3.6 really?
Getting lost with which Py version is tested where, but I thought we wanted to have:
- Py2.7 (legacy)
- Py3.5 (min  Py3 support)
- Py3.8 (max Py3 support: note that 3.9 is not official but the images for 3.9 are not created afaik).

Reply:
This is unchanged, the only change is that I split the big job into 2.

We can modify the Python version in another PR to keep it separated.

Reply:
ok, I don't see Py 3.5 tested anywhere so I guess that 3.6 is min Py version support now. Did it change recently?

Reply:
looks like this minimal version support was changed by @honza last month for xgboost wheel.
Just want us to be aware that we can't claim official support for 3.5 anymore if we don't run any test.

> We can modify the Python version in another PR to keep it separated.

agree, just noticed this because of the highlighting

Reply:
@sebhrusen I think we should have the Python version testing strategy written somewhere - in simple terms so that we can cross-check that the build system is actually doing that.

Also instead of using "pythonVersion: '3.6'" we could use "pythonVersion: PY_MOST_COMMON" / PY_MINIMAL_3 to be more expressive and have it all at one place.

Reply:
yes, good suggestion, same with R.

Reply:
It is complicated right now - at least what we want to accomplish should be written so that anyone can understand and don't change anything that is intentional.